pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -621,6 +622,115 @@ func TestNodeUpdate(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	groupIP := net.ParseIP("224.3.4.5")
+	numEvents := 250
+	var wg sync.WaitGroup
+	wg.Add(4)
+	// Below func adds local group join events.
+	go func() {
+		defer wg.Done()
+		LeastSignificantByteArr := rand.Perm(numEvents)
+		for i := 0; i < len(LeastSignificantByteArr); i++ {
+			c.groupEventCh <- &mcastGroupEvent{
+				group: groupIP,
+				eType: groupJoin,
+				time:  time.Now(),
+				iface: &interfacestore.InterfaceConfig{
+					Type:          interfacestore.ContainerInterface,
+					InterfaceName: fmt.Sprintf("local-interfaceName-%d", i),
+					OVSPortConfig: &interfacestore.OVSPortConfig{
+						OFPort: int32(i),
+					},
+				},
+			}
+		}
+	}()
+	// Below func adds local group leave events.
+	go func() {
+		defer wg.Done()
+		LeastSignificantByteArr := rand.Perm(numEvents)
+		for i := 0; i < len(LeastSignificantByteArr); i++ {
+			c.groupEventCh <- &mcastGroupEvent{
+				group: groupIP,
+				eType: groupLeave,
+				time:  time.Now(),
+				iface: &interfacestore.InterfaceConfig{
+					Type:          interfacestore.ContainerInterface,
+					InterfaceName: fmt.Sprintf("local-interfaceName-%d", i),
+					OVSPortConfig: &interfacestore.OVSPortConfig{
+						OFPort: int32(i),
+					},
+				},
+			}
+		}
+	}()
+	// Below func adds remote group join events.
+	go func() {
+		defer wg.Done()
+		LeastSignificantByteArr := rand.Perm(numEvents)
+		for i := 0; i < len(LeastSignificantByteArr); i++ {
+			c.groupEventCh <- &mcastGroupEvent{
+				group: groupIP,
+				eType: groupJoin,
+				time:  time.Now(),
+				iface: &interfacestore.InterfaceConfig{
+					Type:          interfacestore.TunnelInterface,
+					InterfaceName: fmt.Sprintf("remote-interfaceName-%d", i),
+					OVSPortConfig: &interfacestore.OVSPortConfig{
+						OFPort: int32(i),
+					},
+				},
+				srcNode: net.ParseIP(fmt.Sprintf("10.20.30.%d", LeastSignificantByteArr[i]+2)),
+			}
+		}
+	}()
+	// Below func adds remote group leave events.
+	go func() {
+		defer wg.Done()
+		LeastSignificantByteArr := rand.Perm(numEvents)
+		for i := 0; i < len(LeastSignificantByteArr); i++ {
+			c.groupEventCh <- &mcastGroupEvent{
+				group: groupIP,
+				eType: groupLeave,
+				time:  time.Now(),
+				iface: &interfacestore.InterfaceConfig{
+					Type:          interfacestore.TunnelInterface,
+					InterfaceName: fmt.Sprintf("remote-interfaceName-%d", i),
+					OVSPortConfig: &interfacestore.OVSPortConfig{
+						OFPort: int32(i),
+					},
+				},
+				srcNode: net.ParseIP(fmt.Sprintf("10.20.30.%d", LeastSignificantByteArr[i]+2)),
+			}
+		}
+	}()
+	for i := 0; i < numEvents; i++ {
+		mockIfaceStore.EXPECT().GetInterfaceByName(fmt.Sprintf("local-interfaceName-%d", i)).Return(&interfacestore.InterfaceConfig{
+			Type:          interfacestore.ContainerInterface,
+			InterfaceName: fmt.Sprintf("local-interfaceName-%d", i),
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				OFPort: int32(i),
+			},
+		}, true).AnyTimes()
+	}
```

Comment:
It's contradictory to assert a function is called "anytimes" many times.
If you want accurate assertion, it shouldn't use Anytimes.
If you want it to be called anytimes, no need to have a for loop here.